### DISCUSSION TITLE
Two Rows?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [FelixKratz/SketchyBar#308] [Q&A] [answered]
Is it possible to make Sketchybar display things in *two* rows of items instead of one? For some setups like vertical monitors or my personal "menubar in the corner" this would be very useful

Gone through the docs and it does not look like it, but maybe I missed sth?

**Comment by FelixKratz** [comment, marked as answer]
I think the popup system might be what you are looking for, i.e. you could easily create a second row of items in a popup. The popup supports vertical and horizontal layouts and can accommodate as many items as you like: https://felixkratz.github.io/SketchyBar/config/popups

An example from my dotfiles:

![](https://user-images.githubusercontent.com/22680421/201390348-2b3d038d-24fc-448c-b5e5-827275303295.png)


I have planned a reimplementation of the popup system in the future, because *some* things don’t work as they would in the regular bar and I think I would like the popups to be more like a fully fledged attached secondary bar. This will also be the perfect opportunity to rework  the left and right bar positions for the vertical layout.

**Reply by chrisgrieser** [reply]
thanks, I'll have look!

**Reply by chrisgrieser** [reply]
okay, making this work felt a bit hacky (e.g., attaching everything to the clock, even though the popups have nothing to do with the clock 🥴 ), but I got it working. Thanks a ton!
<img width="275" alt="Pasted image 2023-01-02 21 37 21" src="https://user-images.githubusercontent.com/73286100/210276363-c8a3d121-d0a4-4001-b144-3d7a4de7ce62.png">

**Reply by FelixKratz** [reply]
It feels hacky, because it kind of is. Popup menus are designed for additional information that is *not* always present. Do you have an idea for a better way to achieve this syntactically? Because technically this is very easy to achieve, but I dont know how the configuration structure would best look like fpr a more versatile „widget“ system.

**Reply by chrisgrieser** [reply]
well, essentially, you would achieve that via two steps:
1) "generalize" the idea of a menubar to a rectangle covering some portion of your screen. The menubar would only be the special case of being a thin, full-width rectangle placed at the very top of the screen. The rectangle could simply be defined via x/y + width/height.
2) instead of having text/content flowing in one line, allow for multiple lines. dealing with overflow/wrapping would probably be the more tricky part, but if you leave management of item locations in the hands of the user for starters, a multi-line model could suffice?

**Reply by FelixKratz** [reply]
> well, essentially, you would achieve that via two steps:
> 
> 1. "generalize" the idea of a menubar to a rectangle covering some portion of your screen. The menubar would only be the special case of being a thin, full-width rectangle placed at the very top of the screen. The rectangle could probably be defined via x/y + width/height.
> 2. instead of having text/content flowing in one line, allow for multiple lines. dealing with overflow/wrapping would probably be the more tricky part, but if you leave management of item locations in the hands of the user for starters, a multi-line model could suffice?

Technically, all items could be positioned at arbitrary coordinates on the screen already, because each item has its own window. But I think this project will focus on the traditional bar concept while trying to make it configurable and as easy to use as possible. There exists some sweet spot between configurability and usability that I am trying to achieve for advanced users and I fear that making these kinds of configurational degrees of freedom available will lead to a drastic decrease of usability.

**Reply by chrisgrieser** [reply]
totally true. Maybe a just adding a second row would be inside that scope? 🙏

**Reply by FelixKratz** [reply]
There is something in the making for actual multiple bars: https://github.com/FelixKratz/SketchyBar/issues/353

**Reply by chrisgrieser** [reply]
woah, I'll have a look at the weekend, thanks for letting me know
